public/app/features/plugins/admin/components/PluginDetailsBody.tsx

```diff
@@ -1,25 +1,26 @@
 import React from 'react';
 import { css, cx } from '@emotion/css';
 
-import { AppPlugin, GrafanaTheme2 } from '@grafana/data';
+import { AppPlugin, GrafanaTheme2, UrlQueryMap } from '@grafana/data';
 import { useStyles2 } from '@grafana/ui';
 
-import { CatalogPlugin, PluginTabLabels } from '../types';
+import { CatalogPlugin, PluginTabIds } from '../types';
 import { VersionList } from '../components/VersionList';
 import { usePluginConfig } from '../hooks/usePluginConfig';
 import { AppConfigCtrlWrapper } from '../../wrappers/AppConfigWrapper';
 import { PluginDashboards } from '../../PluginDashboards';
 
 type Props = {
-  tab: { label: string };
   plugin: CatalogPlugin;
+  queryParams: UrlQueryMap;
```

Comment:
**Question:** couldn't we use [`useParams()`](https://reactrouter.com/web/api/Hooks/useparams) instead of pushing down the props here?

Reply:
I was in two minds about pushing them down vs using a hook. I went with pushing them down as they're being used in the parent component and whilst there is now coupling I can't see this component being used elsewhere so felt it was ok.

Looks like the `useParams` hook gets the params from the path. e.g. `/plugin/:pluginId`. However we do have a `useQueryParams` hook [here](https://github.com/grafana/grafana/blob/main/public/app/core/hooks/useQueryParams.ts) that we can use instead.

Reply:
I see, thanks for the explanation bro!

I think it is good as it is for now, we can always come back and change it if we would find the coupling holding us back.